src/dev/flang/be/jvm/runtime/Runtime.java

```diff
@@ -711,6 +724,190 @@ public static byte[] fuzion_sys_env_vars_get0(Object d)
+  {
+    Object result;
+    String clazz = null;
+
+    try
+      {
+        Class cl = thiz.getClass();
+        Field f = cl.getDeclaredField(field);
+        result = f.get(thiz);
+      }
+    catch (IllegalAccessException | NoSuchFieldException e)
+      {
+        Errors.fatal(e.toString()+" when calling fuzion.java.get_static_field for field "+clazz+"."+field);
+        result = null;
+      }
+
+    return result;
+  }
+
+
+  public static Object fuzion_java_call_v0(String clName, String name, String sig, Object thiz, Object[] args)
+  {
+    if (PRECONDITIONS) require
+      (clName != null);
+
+    Method m = null;
+    var p = JavaInterface.getPars(sig);
+    if (p == null)
+      {
+        Errors.fatal("could not parse signature >>"+sig+"<<");
+      }
+    Class cl;
+    try
+      {
+        cl = Class.forName(clName);
+      }
+    catch (ClassNotFoundException e)
+      {
+        Errors.fatal("ClassNotFoundException when calling fuzion.java.call_virtual for class " +
+                           clName + " calling " + name + sig);
+        cl = Object.class; // not reached.
+      }
+    try
+      {
+        m = cl.getMethod(name, p);
+      }
+    catch (NoSuchMethodException e)
+      {
+        Errors.fatal("NoSuchMethodException when calling fuzion.java.call_virtual calling " +
+                           (cl.getName() + "." + name) + sig);
+      }
+    return invoke(m, thiz, args);
+  }
+
+  static interface ReflectionInvoker
+  {
+    Object invoke() throws InvocationTargetException, IllegalAccessException, InstantiationException;
+  }
+
+
+  private static Object invokeAndWrapException(ReflectionInvoker invoke)
+  {
+    Object res;
+    try
+      {
+        res = invoke.invoke();
+      }
+    catch (InvocationTargetException e)
+      {
+        ((FuzionThread)Thread.currentThread())._thrownException = e.getCause();
+        res = _JAVA_ERROR_;
+      }
+    catch (IllegalAccessException | InstantiationException e)
+      {
+        ((FuzionThread)Thread.currentThread())._thrownException = e;
+        res = _JAVA_ERROR_;
+      }
+    return res;
+  }
+
+  private static Object invoke(Method m, Object thiz, Object[] args)
+  {
+    return invokeAndWrapException(()->m.invoke(thiz, args));
+  }
+
+
+  public static Object fuzion_java_call_s0(String clName, String name, String sig, Object[] args)
+  {
+    if (PRECONDITIONS) require
+      (clName != null);
+
+    Method m = null;
+    Constructor co = null;
+    var  p = JavaInterface.getPars(sig);
+    if (p == null)
+      {
+        Errors.fatal("could not parse signature >>"+sig+"<<");
+      }
+    Class cl;
+    try
+      {
+        cl = Class.forName(clName);
+      }
+    catch (ClassNotFoundException e)
+      {
+        Errors.fatal("ClassNotFoundException when calling fuzion.java.call_static for class " +
+                           clName + " calling " + name + sig);
+        cl = Object.class; // not reached.
+      }
```

Comment:
Except for the error message, this code is a copy of the same code in `fuzion_java_call_v0`, can you extract it into a helper method?